authappliance/menu.py

```diff
@@ -17,10 +17,30 @@
 """
 text dialog based setup tool to configure the privacyIDEA basics.
 """
-
+from __future__ import print_function
 import locale
 import argparse
 import sys
+import os
+from traceback import print_exc
+
+# If the ``privacyidea`` Python package cannot be imported normally,
+# append the virtualenv site-packages of the privacyIDEA 3.x distribution
+# packages to the search path and try again.
+PRIVACYIDEA_SITE_PACKAGES = '/opt/privacyidea/lib/python2.7/site-packages/'
```

Comment:
Currently this could would not run on Python 3.

Reply:
You mean this stops working if we upgrade our privacyIDEA packages to Python 3, because the search path is specific to Python 2.7? That's true, we would need to create a new appliance package then as well. But I think this would also be the case if we use ``.pth`` files or manually patch the appliance code in the Ubuntu packages.

Hm, when we upgrade our privacyIDEA packages to Python 3, wouldn't we also upgrade the appliance packages to Python 3 anyway?

Reply:
We could make this dependent of the current python version.

Reply:
OK, I see now. 
I though we would have problems, if s.o. would install it with Python 3.
But if s.o. would install it with Python 3, this s.o. would do this either in-system or in a shared virtualenv and then the import from line 32 would work.

If **we** change in our packages privacyidea to run on Python 3 we would also need to compile a new pi-appliance package (with a Python 3 link).
So this is fine.
(Unless, @plettich you want to replace "python2.7" in the path with more logic (which at least currently is imho not necessary)

Reply:
To avoid extra (and error prone) editing, we can use something like this:
```python
import platform
site_version = 'python' + '.'.join(platform.python_version_tuple()[0:2])
PRIVACYIDEA_SITE_PACKAGES = '/opt/privacyidea/lib/' + site_version + '/site-packages/'
```

Reply:
This will save us some coding changes when we migrate privacyIDEA + Appliance to Python 3, we can do that!

However, this won't work if we have a privacyIDEA package based on Python 3 with an Appliance package based on Python 2, because ``platform.python_version_tuple()`` will refer to Python 2, whereas the correct privacyIDEA site-packages will be at ``/opt/privacyidea/lib/python3.7/site-packages/`` (or so). Same situation with a privacyIDEA package based on Python 2 and an Appliance package based on Python 3.

But I'm not sure if we should support these scenarios. :)

Reply:
I think a Python 2 appliance is very unlikely to work with y Python 3 privacyIDEA anyways.
So I think @plettich 's approach is fine.

Reply:
OK, sounds good! I added @plettich's snippet in 5e57211 :)